src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -977,12 +968,12 @@ public static SocketError Receive(SafeCloseSocket handle, byte[] buffer, int off
         {
```

Comment:
Couldn't we get rid of this method, and similar, and just have callers use Span?  Maybe beyond the scope of this change...

Reply:
I actually did that and then backed it out.  The problem there is what happens if the operation doesn't complete synchronously and we need to fall back to the async path.  If we have the original array, then we can store that on the heap, but if we only have the span, we have to pin the span, get a pointer for it, and then do that special pointer thing, meaning we end up pinning when we otherwise wouldn't need to.

Reply:
Yeah, this is the same reason we have Memory<byte> versions of sync ops on the Linux SocketAsyncContext, right?

For what it's worth, we could make this better by making the sync ops in SocketAsyncContext dispatch back to the calling thread instead of the thread pool.  This is arguably better anyway.  But it's a nontrivial change and not sure it's worth doing anytime soon.

Reply:
> Yeah, this is the same reason we have Memory versions of sync ops on the Linux SocketAsyncContext, right?

Yup